src/System.Management.Automation/engine/regex.cs

```diff
@@ -233,33 +222,34 @@ internal static string Escape(string pattern, char[] charsNotToEscape)
             Span<char> temp = pattern.Length < StackAllocThreshold ? stackalloc char[pattern.Length * 2 + 1] : new char[pattern.Length * 2 + 1];
```

Comment:
Assuming you intended to write `pattern.Length * 2 < StackAllocThreshold` 😛, use a constant for the stack allocation length and slice later:

```suggestion
            int lengthRequired = pattern.Length * 2 + 1;
            Span<char> temp = lengthRequired <= StackAllocThreshold ? stackalloc char[StackAllocThreshold] : new char[lengthRequired];
```